prow/apis/prowjobs/v1/types_test.go

```diff
@@ -0,0 +1,374 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"reflect"
+	"testing"
+	"time"
+)
+
+func TestDecorationDefaulting(t *testing.T) {
+	truth := true
+	lies := false
+	defaults := &DecorationConfig{
+		Timeout:     1 * time.Minute,
+		GracePeriod: 10 * time.Second,
+		UtilityImages: &UtilityImages{
+			CloneRefs:  "clonerefs",
+			InitUpload: "iniupload",
+			Entrypoint: "entrypoint",
+			Sidecar:    "sidecar",
+		},
+		GCSConfiguration: &GCSConfiguration{
+			Bucket:       "bucket",
+			PathPrefix:   "prefix",
+			PathStrategy: PathStrategyLegacy,
+			DefaultOrg:   "org",
+			DefaultRepo:  "repo",
+		},
+		GCSCredentialsSecret: "secretName",
+		SSHKeySecrets:        []string{"first", "second"},
+		SSHHostFingerprints:  []string{"primero", "segundo"},
+		SkipCloning:          &truth,
+	}
+
+	var testCases = []struct {
+		name     string
+		provided *DecorationConfig
+		expected *DecorationConfig
```

Comment:
As exciting as copy-paste inheritance may be, `expected func(left, right *DecorationConfig) *DecorationConfig` will make these tests much easier to read

Reply:
Done.